seed/seedwriter/seed16.go

```diff
@@ -108,6 +111,35 @@ func (pol *policy16) checkBase(info *snap.Info, availableSnaps *naming.SnapSet)
 	return fmt.Errorf("cannot add snap %q without also adding its base %q explicitly", info.SnapName(), info.Base)
 }
 
+func (pol *policy16) needsImplicitSnaps(availableSnaps *naming.SnapSet) (bool, error) {
+	// do we need to add implicitly either snapd (or core)
+	hasCore := availableSnaps.Contains(naming.Snap("core"))
+	if len(pol.needsCore) != 0 && !hasCore {
+		// XXX warning on Core 18
+		return true, nil
+	}
+
+	if len(pol.needsCore16) != 0 && !hasCore {
```

Comment:
Should this check for core16 too?
```
hasCore16 := availableSnaps.Contains(naming.Snap("core16"))
if len(pol.needsCore16) != 0 && (!hasCore || !hasCore16) { ..
```
Although not sure whether seed with core16 would actually be useful.

Reply:
so far core16 never made it past beta, and at this point is a unclear when/if that happens.  The *implicit* code here corresponds to this code in image.go:

https://github.com/snapcore/snapd/blob/master/image/image.go#L485
https://github.com/snapcore/snapd/blob/master/image/image.go#L503

Reply:
Makes sense, we can always revisit if there's ever a need to do so.